source/common/statistic_impl.h

```diff
@@ -85,6 +110,7 @@ class InMemoryStatistic : public StatisticImpl {
     return streaming_stats_->resistsCatastrophicCancellation();
   }
   uint64_t significantDigits() const override { return streaming_stats_->significantDigits(); }
+  StatisticPtr createNewInstance() const override { return std::make_unique<InMemoryStatistic>(); };
```

Comment:
+1 on the clever elimination of the factory. 

When I saw this being called, I assumed createNewInstance() would create a copy of the existing object -- the most obvious explanation for why it would be an instance method on an existing object. What about createNewInstanceOfSameType() to clarify the meaning?